.travis.yml

```diff
@@ -18,7 +18,7 @@ before_install:
   - cmake . && make && sudo make install && cd ..
 
 install:
-  - mkdir build && cd build && cmake .. && make -j2
+  - mkdir build && cd build && cmake -DSHOW_COMPILE_WARNINGS=ON .. && make -j2
```

Comment:
Hmm, could we simplify by just calling `cmake -DCMAKE_CXX_FLAGS='-Wall -Wpedantic -Wextra -Werror' ..`?  Or is there a compelling reason to add the `SHOW_COMPILE_WARNINGS` option? (maybe it makes life easier downstream?)

Reply:
Definitely. Though, how would you feel if I just mirrored MLPACK's setup?

https://github.com/mlpack/mlpack/blob/3389a40ecd686a7e1e01a63b475b042e804440ff/CMakeLists.txt#L65-L78

Reply:
That's totally fine with me.  We can use the warning flags on every build if you like (except `-Werror` I would say, I would agree with Conrad that that can be really annoying).  But perhaps it is  good to enable `-Werror` just for the Travis build (via `CMAKE_CXX_FLAGS`) so we avoid committing any code that causes warnings.